tests/xharness/XmlResultParser.cs

```diff
@@ -0,0 +1,267 @@
+﻿using System;
+using System.IO;
+using System.Xml;
+
+namespace xharness {
+	public static class XmlResultParser {
+
+		public enum XmlResultType {
+			TouchUnit,
+			NUnit,
+			xUnit,
+		}
+
+		// test if the file is valid xml, or at least, that can be read it.
+		public static bool IsXml (string filePath)
+		{
+			if (!File.Exists (filePath))
+				return false;
+
+			using (var stream = File.OpenText (filePath)) {
+				string line;
+				while ((line = stream.ReadLine ()) != null) { // special case when get got the tcp connection
+					if (line.Contains ("ping"))
+						continue;
+					return line.StartsWith ("<", StringComparison.Ordinal);
+				}
+			}
+			return false;
+		}
+
+		public static XmlResultType GetXmlType (string filePath)
```

Comment:
can't this be merged inside `IsXml` ?
we're parsing (part of) the same file twice

instead you could have `bool IsValidXml (string path, out XType type)`